src/main/java/org/apache/commons/compress/utils/FileNameUtils.java

```diff
@@ -50,6 +51,19 @@ public static String getExtension(final String filename) {
         return name.substring(extensionPosition + 1);
     }
 
+    public static String getExtensionFrom(final Path path) {
```

Comment:
Make this private or package private. The fewer public or protected method the better as they will have to be maintained "forever". Also needs a comment.